pkg/server/protoutils.go

```diff
@@ -140,37 +143,32 @@ func createUnsignedTx(
 	return &common.Envelope{Payload: paylBytes}, nil
 }
 
-func getRandomNonce() ([]byte, error) {
-	key := make([]byte, 24)
-
-	_, err := rand.Read(key)
-	if err != nil {
-		return nil, errors.Wrap(err, "error getting random bytes")
-	}
-	return key, nil
-}
-
-func getChannelHeaderFromSignedProposal(signedProposal *peer.SignedProposal) (*common.ChannelHeader, error) {
+func getChannelAndChaincodeFromSignedProposal(signedProposal *peer.SignedProposal) (string, string, error) {
```

Comment:
The channel name is a top-level property of the ProposedTransaction so we don't need to deserialise and dig into the proposal to get it. I guess we should either:
1. Also move the chaincode name to the top-level of the ProposedTransaction, which means we can save a whole bunch of expensive deserialisation on the proposals within the Gateway; or
2. Just take the hit on deserialising the proposal and remove any ProposedTransaction properties that are duplicates of content in the proposal for simplicity.

But we could do that work outside of this PR, just as long as we actually do it.

Reply:
ok, this will all need revisiting anyway when we decide exactly what goes into the top-level message